lnrpc/routerrpc/subscribe_events.go

```diff
@@ -68,7 +68,19 @@ func rpcHtlcEvent(htlcEvent interface{}) (*HtlcEvent, error) {
 		}
 
 		key = e.HtlcKey
-		eventType = e.HtlcEventType
+		eventType = &e.HtlcEventType
+		timestamp = e.Timestamp
+
+	case *htlcswitch.FinalHtlcEvent:
+		event = &HtlcEvent_FinalHtlcEvent{
```

Comment:
Any reason this event don't have an associated `eventType`. Feels like an alternative (better?) option would be to extend the `SettleEvent` with a `final` field.

Reply:
It is just much cleaner to have uniform handling of the event, and I would say it makes no sense for `SettleEvent` to have an event type but not this one? (it is basically the same).

Reply:
>Feels like an alternative (better?) option would be to extend the SettleEvent with a final field.

Extending `SettleEvent` wouldn't work for the final fail case. And for a failure, there are two events: `LinkFail` and `ForwardingFail`. Not sure if extending all of them is a good option.

>It is just much cleaner to have uniform handling of the various events, and I would say it makes no sense for SettleEvent to have an event type but not this one? (it is basically the same).

I didn't want to add a dedicated subscriber api for final htlc notification and decided that the existing htlc notifier api is close enough to add it there. The use case, at least mine, is quite different though. The htlc notifier is intended to keep a (lossy) eye on traffic, whereas the final htlc notification is needed to get a definite answer to what happened to an htlc without relying on polling `LookupHtlc`. The additional context of whether the htlc was a forward or exit hop receive isn't relevant.

I understand what you're saying about making it uniform though. I checked what it would take and it seems that it isn't that straight-forward. In the on-chain resolvers and also off-chain when the resolutions get locked in fully, the info that is required isn't readily available.

What I'd propose is to stick to a separate `FinalHtlcEvent` and leave the addition of `eventType` until there is an actual need for that. At that point, it can be added as a non-breaking change.

Reply:
Yeah, agreed.